src/index.js

```diff
@@ -17,7 +20,25 @@ class ProjectionGridReact extends React.Component {
   }
 
   render() {
-    const model = this.core.compose({ config: _.omit(this.props, 'projections') });
+    const { classes, data, caption, cols, primaryKey, sorting, tfoot } = this.props;
+
+    const model = this.core.compose({
+      config: {
+        classes: utils.compact([...classes, this.props.className]),
+        data,
+        caption,
+        cols,
+        primaryKey,
+        sorting,
+        tfoot,
+      },
+      state: this.gridState,
```

Comment:
Seems it's OK to use `this.state`. If we call `this.setState`, as long as the state is not updated, the `render` method is not called. It means, whenever the `render` is called, the `state` is up to date.

Reply:
We the `gridState` or the sync state is only used in reducers. So the logic could be
```js
{
  ...
  state: this.state,
  dispatch: (reducer, ...args) => {
    const state = this.gridState = reducer(this.gridState, ...args);
    this.setState(state);
    return state;
  },
}
```